### PR TITLE
Qtuan1401/add/SignupPage

### PR DESCRIPTION
**Pull request by qtuan1401**
Add signup page, add new image and add new route into App.js

### CODE REVIEW

**Comment by kc97ble** [comment]
this branch should not contain commits from enter page.

File: src/app/components/LoginPage/failed.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import styles from './failed.scss';
+
+import CloseButton from '../../../assets/images/XMLID 1429.png';
```
Comment (kc97ble) [inline src/app/components/LoginPage/failed.js:5]: please rename the file

e.g. "close-button.png"

File: src/app/components/LoginPage/index.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  handleSubmit(event) {
+    event.preventDefault();
+    const { username, userPassword } = this.state;
+    const name = String(username);
+    const pass = String(userPassword);
+    if (name !== userInfo.username[0] || pass !== userInfo.userPassword[0]) {
+      isRedirect = 2;
+    } else {
+      isRedirect = 1;
+    }
```
Comment (kc97ble) [inline src/app/components/LoginPage/index.js:55]: Don't use global variable.

React does not watch global variables. Even if `isRedirect` is changed, there is no guarantee that `render` function will be triggered.

Use React's state for this purpose.

File: src/app/components/LoginPage/login.scss [hunk 3]
```diff
@@ -0,0 +1,148 @@
+@import url('https://fonts.googleapis.com/css?family=Mulish');
+
+:local {
+  .popup {
```
Comment (kc97ble) [inline src/app/components/LoginPage/login.scss:4]: rename to `overlay` or `popupOverlay` or `overlayBackground`

File: src/app/components/LoginPage/login.scss [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    position: fixed;
+    top: 0px;
+    left: 0px;
+    width: 1920px;
+    height: 1080px;
```
Comment (kc97ble) [inline src/app/components/LoginPage/login.scss:9]: what if user's screen resolution is not 1920 x 1080?

to make the element full screen, use:
```
    position: fixed;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
```

File: src/app/components/LoginPage/login.scss [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+  }
+
+  .login {
+    position: absolute;
+    width: 600px;
+    height: 352px;
+    left: 660px;
+    top: 96px;
+    background: #FFFFFF;
+    box-shadow: 0px 0px 12px rgba(188, 188, 188, 0.25);
```
Comment (kc97ble) [inline src/app/components/LoginPage/login.scss:22]: Avoid `position: absolute` for simple components.

There are no good reasons to use `position: absolute` here.

File: src/app/components/SignupPage/index.js [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+let chkPassword = true;
+const testPassword = new RegExp('^(?=.*[0-9])(?=.*[a-zA-Z]).{8,}$');
+const testEmail = new RegExp('^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+[a-zA-Z0-9-.]+$');
+let isMatchPassword = true;
+let chkFullname = true;
+let chkUsername = true;
+let chkEmail = true;
+let chkAddress = true;
+let chkSchool = true;
+let chkAgreed = true;
```
Comment (kc97ble) [inline src/app/components/SignupPage/index.js:21]: Don't use global variables.

(global consts are okay)

File: src/app/components/SignupPage/index.js [hunk 7]
```diff
@@ -0,0 +1,294 @@
...
+                <p>Để tham gia FYT Code Cup, hãy đăng ký trước 19:30 ngày 09/10/2020 bạn nhé!</p>
+              </div>
+            </div>
+            <div className={styles.content}>
+              <div className={styles.fullname}>
+                <small>Họ và tên</small> <br />
+                <input type="text" name="fullname" onChange={this.handleChange} />
+                <div className={styles.errorMessage}>
+                  <ShowError chk={chkFullname} s1="" s2="Xin vui lòng điền thông tin" />
+                </div>
```
Comment (kc97ble) [inline src/app/components/SignupPage/index.js:158]: Abstract these input fields into components.

You are having a lot of duplications here.

Comment (kc97ble) [reply src/app/components/SignupPage/index.js:158]: e.g.

```
function InputText({ label, name, onChange, value }) {
...
}
```

File: src/app/components/LoginPage/index.js [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+    this.forceUpdate();
+  }
+
+  handleChange(event) {
+    const name = [event.target.name];
+    const value = [event.target.value];
+    this.setState({
+      [name]: value,
```
Comment (hnngo) [inline src/app/components/LoginPage/index.js:63]: is it wrong, so name and value here are Array?

File: src/app/components/SignupPage/failed.scss [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+      line-height: 25px;
+      
+      display: flex;
+      align-items: center;
+      text-align: center;
+      justify-content: center;
+      color: rgba(0, 0, 0, 0.8);
+    }
+  
+    button {
```
Comment (hnngo) [inline src/app/components/SignupPage/failed.scss:60]: Avoid general element tag

**Review by hnngo — CHANGES_REQUESTED** [review]

File: src/app/components/EnterPage/enter.scss [hunk 10]
```diff
@@ -0,0 +1,73 @@
+@import url('https://fonts.googleapis.com/css?family=Mulish');
```
Comment (hnngo) [inline src/app/components/EnterPage/enter.scss:1]: Remove this line, font has been added once in html file

**Review by hnngo — CHANGES_REQUESTED** [review]

**Pull request closed without merging**